src/makeCalculator.js

```diff
@@ -3,8 +3,44 @@
 /**
  * @return {object}
  */
-function makeCalculator() {
-  // write code here
+function makeCalculator(operate) {
+  const obj = {
+    result: 0,
+
+    reset: function () {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate: function (callback, number) {
+      this.result = callback(this.result, number);
+
+      return this;
+    },
+
+    add: function (currentResult, number) {
+      return currentResult + number;
+    },
+
+    subtract: function (currentResult, number) {
+      return currentResult - number;
+    },
+
+    multiply: function (currentResult, number) {
+      return currentResult * number;
+    },
+
+    divide: function (currentResult, number) {
+      if (number === 0) {
+        return 'cannot be divided by zero';
+      }
+
+      return currentResult / number;
```

Comment:
Use the method shorthand syntax for defining methods in an object. Replace 'divide: function (currentResult, number) {' with 'divide(currentResult, number) {'. Also, instead of returning a string when dividing by zero, it would be better to handle the error or throw an exception.